### PR TITLE
Ontology replace

### PR DESCRIPTION
**Pull request by Foundkim**
Hello Christophe, I,ve made the job you ask about the ontology.py and account-ontology.csv files

### CODE REVIEW

**Review by phe-sto — CHANGES_REQUESTED** [review]
Please do not mix up PR. There should modify just one feature at a time to be easily reviewable.
Could place the APE filter modification back to the corresponding PR.

**Comment by Foundkim** [comment]
This pull request contains just ontology.py and extract_bundle.py modifcations. I will push the APE CODE filter in the right branch

File: python/enthic/app.py [hunk 1]
```diff
@@ -178,21 +178,23 @@ def pre_cast_integer(probe):
     return str(probe) if probe.__class__ is int else probe if probe.isnumeric() is True else None
 
 
-def result_array(probe, limit, offset=0):
+def result_array(probe, limit, apecode, offset=0):
```
Comment (phe-sto) [inline python/enthic/app.py:181]: This modification relates to the ape code filter.

File: python/enthic/app.py [hunk 2]
```diff
@@ -178,21 +178,23 @@ def pre_cast_integer(probe):
...
         companies = fetchall("""SELECT siren, denomination, ape, postal_code, town
-                        FROM identity WHERE siren = %s
-                        OR denomination LIKE %s
-                        OR MATCH(denomination) AGAINST (%s IN NATURAL LANGUAGE MODE)
-                        LIMIT %s OFFSET %s;""", (pre_cast_integer(probe), "{0}%".format(probe),
-                                                 "{0}%".format(probe), limit, offset))
+                        FROM identity WHERE (siren = %s and ape regexp %s)
+                        OR (denomination LIKE %s and ape regexp %s)
+                        OR (MATCH(denomination) AGAINST (%s IN NATURAL LANGUAGE MODE) and ape regexp %s)
+                        LIMIT %s OFFSET %s;""", (pre_cast_integer(probe),"^{0}".format(apecode),
```
Comment (phe-sto) [inline python/enthic/app.py:195]: This modification relates to the ape code filter.

File: python/enthic/app.py [hunk 3]
```diff
@@ -216,22 +218,26 @@ def search():
         # WRONG TYPE
         if json_data["limit"].__class__ is not int and \
                 (json_data["probe"].__class__ is not str and json_data[
-                    "probe"].__class__ is not int):
+                    "probe"].__class__ is not int) and \
+                        json_data['ape'].__class__ is not int:
```
Comment (phe-sto) [inline python/enthic/app.py:222]: This modification relates to the ape code filter.

File: python/enthic/app.py [hunk 4]
```diff
@@ -216,22 +218,26 @@ def search():
...
             return ErrorJSONResponse(
-                "Value limit must be a string or integer and limit an integer."
+                "Value limit must be a string or integer and limit an integer\
+                and value ape an integer"
             )
         elif json_data["limit"].__class__ is not int:
             return ErrorJSONResponse("Value limit must be an integer.")
         elif json_data["probe"].__class__ is not str and json_data["probe"].__class__ is not int:
             return ErrorJSONResponse("Value probe must be a string or integer.")
+        elif json_data["ape"].__class__ is not int:
```
Comment (phe-sto) [inline python/enthic/app.py:231]: This modification relates to the ape code filter.

File: python/enthic/app.py [hunk 5]
```diff
@@ -216,22 +218,26 @@ def search():
...
+            return ErrorJSONResponse("Value ape must be an integer.")
         ########################################################################
         # WRONG LIMIT
         elif json_data["limit"] > 10000:
             return ErrorJSONResponse("Value limit is 10000 maximum.")
         ########################################################################
         # CORRECT JSON
         else:
-            results = result_array(json_data["probe"], json_data["limit"])
+            results = result_array(json_data["probe"], json_data["limit"], json_data["ape"])
```
Comment (phe-sto) [inline python/enthic/app.py:240]: This modification relates to the ape code filter.

File: python/enthic/app.py [hunk 6]
```diff
@@ -216,22 +218,26 @@ def search():
         # WRONG TYPE
         if json_data["limit"].__class__ is not int and \
                 (json_data["probe"].__class__ is not str and json_data[
-                    "probe"].__class__ is not int):
+                    "probe"].__class__ is not int) and \
+                        json_data['ape'].__class__ is not int:
             return ErrorJSONResponse(
-                "Value limit must be a string or integer and limit an integer."
+                "Value limit must be a string or integer and limit an integer\
```
Comment (phe-sto) [inline python/enthic/app.py:224]: This modification relates to the ape code filter.

File: python/enthic/app.py [hunk 7]
```diff
@@ -288,31 +296,31 @@ def page_search():
...
             "@id": request.full_path,
             "@type": "PartialCollectionView",
-            "first": '%s?page=1&per_page=%d&probe=%s' % (request.path,
-                                                         per_page, probe),
-            "last": '%s?page=%d&per_page=%d&probe=%s' % (request.path,
-                                                         last_per_page,
-                                                         per_page,
-                                                         probe)},
+            "first": '%s?page=1&per_page=%d&probe=%s&ape=%s' % (request.path,
+                                                                 per_page, probe, ape),
```
Comment (phe-sto) [inline python/enthic/app.py:300]: This modification relates to the ape code filter.

File: python/enthic/app.py [hunk 8]
```diff
@@ -288,31 +296,31 @@ def page_search():
...
     # MAKE NEXT URL
     if page * per_page + per_page > count:
         obj["view"]["next"] = ''
     else:
-        obj["view"]["next"] = '%s?page=%d&per_page=%d&probe=%s' % (request.path,
-                                                                   page + 2,
-                                                                   per_page,
-                                                                   probe)
+        obj["view"]["next"] = '%s?page=%d&per_page=%d&probe=%s&ape=%s' % (request.path,
+                                                                     page + 2,
```
Comment (phe-sto) [inline python/enthic/app.py:321]: This modification relates to the ape code filter.

File: python/enthic/app.py [hunk 9]
```diff
@@ -257,21 +263,23 @@ def page_search():
...
-        future_list = executor.submit(result_array, probe, per_page, offset=page * per_page)
+        future_list = executor.submit(result_array, probe, per_page, ape, offset=page * per_page)
         count, = fetchone("""SELECT COUNT(*)
-                            FROM identity WHERE siren = %s
-                            OR denomination LIKE %s
-                            OR MATCH(denomination) AGAINST (%s IN NATURAL LANGUAGE MODE)""",
-                          (pre_cast_integer(probe), "{0}%".format(probe), "{0}%".format(probe)))
+                        FROM identity WHERE (siren = %s and ape regexp %s)
+                        OR (denomination LIKE %s and ape regexp %s)
+                        OR (MATCH(denomination) AGAINST (%s IN NATURAL LANGUAGE MODE)
```
Comment (phe-sto) [inline python/enthic/app.py:279]: This modification relates to the ape code filter.

**Review by phe-sto — CHANGES_REQUESTED** [review]
I still have app.py modified with the ape code filter feature.

File: python/enthic/app.py (hunk 9 above)
Comment (Foundkim) [reply python/enthic/app.py:279]: Okay, it's a mistake. I 'll change rectify

**Pull request closed without merging**